src/Aspire.Dashboard/Components/Layout/MainLayout.razor.cs

```diff
@@ -124,8 +130,15 @@ protected override async Task OnAfterRenderAsync(bool firstRender)
         {
             _jsModule = await JS.InvokeAsync<IJSObjectReference>("import", "/js/app-theme.js");
             _shortcutManagerReference = DotNetObjectReference.Create(ShortcutManager);
+            _layoutReference = DotNetObjectReference.Create(this);
             _keyboardHandlers = await JS.InvokeAsync<IJSObjectReference>("window.registerGlobalKeydownListener", _shortcutManagerReference);
+            _textVisualizerHandler = await JS.InvokeAsync<IJSObjectReference>("window.registerOpenTextVisualizerOnClick", _layoutReference);
             ShortcutManager.AddGlobalKeydownListener(this);
+
+            // Get initial theme and then rerender the page to apply it.
+            await ThemeManager.RaiseThemeChangedAsync(await _jsModule.InvokeAsync<string>("getThemeCookieValue"));
+
+            await InvokeAsync(StateHasChanged);
```

Comment:
What is this for? Performance of the initial page load is important. Re-rendering the page after it was rendered will have a negative impact.

Reply:
I looked into what is happening here and see why you're doing this. For consistency with the rest of the app, theme switching should be (as much as possible) a client-side operation. The highlight styling was re-rending the page to add or remove CSS links is very different than what we're doing everywhere else.

I updated the code to use one stylesheet and control colors via variables.

Before I did this, I made a bunch of changes around getting the "effective" theme on the server. It turns out that's no longer required. I'm going to leave it in in case it is ever needed in the future.